tmt/steps/provision/__init__.py

```diff
@@ -1103,6 +1109,59 @@ def requires(cls) -> list['tmt.base.Dependency']:
         """ All requirements of the guest implementation """
         return []
 
+    def acquire_log(self, log_name: str) -> Optional[str]:
+        """
+        Fetch and return content of a log.
+        :param log_name: name of the log.
+        :returns: content of the log.
+        """
+        raise NotImplementedError
+
+    def store_log(
+            self,
+            log_path: Path,
+            log_content: str,
+            log_name: Optional[str] = None) -> None:
+        """
+        Save log content to a file.
+        :param log_path: a path to save into,could be a directory
+        or a file path.
+        :param log_content: content of the log.
+        :param log_name: name of the log, if not set, log_path
+        is supposed to be '/dev/null' or  a file path.
+        """
+        # if log_path is file path or /dev/null
+        if not log_path.is_dir() or str(log_path) == '/dev/null':
+            log_path.write_text(log_content)
+        # log_path is a directory
+        else:
+            if log_name:
+                name_str = log_name
+            else:
+                name_str = 'tmt-guestlog-' + \
```

Comment:
Nah, I wouldn't bother. File name or log name are inputs, it's up to the caller to provide them. If `log_path.is_dir()` is true and `log_name` is not set, let's raise an exception instead of inventing names. If there's no `log_name`, we would end up with a bunch of files with very similar names - when it was clearly a bug in the upper level that did know log names but did not bother to give them to this method.